akka-http/src/main/scala/akka/http/javadsl/unmarshalling/Unmarshaller.scala

```diff
@@ -43,13 +43,23 @@ object Unmarshaller {
   def async[A, B](f: java.util.function.Function[A, CompletionStage[B]]): Unmarshaller[A, B] =
     unmarshalling.Unmarshaller[A, B] { ctx ⇒ a ⇒ f(a).toScala
     }
+  /*
+  //#unmarshaller-creation
+  static <A, B> Unmarshaller<A, B> async(java.util.function.Function<A, java.util.concurrent.CompletionStage<B>> f)
```

Comment:
Could we move these to a Java file, for example under `docs/src/test/java/docs/http/javadsl`?

Reply:
I can maybe move them to a Java interface?

Reply:
Introduce a Java interface used by the Unmarshaller companion object.